shap/explainers/_tree.py

```diff
@@ -1023,8 +1030,12 @@ def __init__(self, model, data=None, data_missing=None, model_output=None):
             self.base_offset = xgb_loader.base_score
             self.objective = objective_name_map.get(xgb_loader.name_obj, None)
             self.tree_output = tree_output_name_map.get(xgb_loader.name_obj, None)
-            if xgb_loader.num_class > 0:
-                self.num_stacked_models = xgb_loader.num_class
+            self.num_stacked_models = xgb_loader.n_trees_per_iter
+            self.cat_feature_indices = xgb_loader.cat_feature_indices
+            best_iteration = getattr(
+                model, "best_iteration", model.num_boosted_rounds() - 1
+            )
+            self.tree_limit = (best_iteration + 1) * self.num_stacked_models
```

Comment:
As an aside, this method seems far too long. It could be good to refactor this as perhaps as series of class methods: `cls.from_xbg_booster()`, `cls.from_xgb_classifier()` and so on. I'll save that for another PR though.